src/core/backbone_router/bbr_local.cpp

```diff
@@ -263,6 +263,7 @@ void Local::HandleBackboneRouterPrimaryUpdate(Leader::State aState, const Backbo
     {
         Reset();
     }
+#if !OPENTHREAD_CONFIG_REFERENCE_DEVICE_ENABLE
     else if (!mIsServiceAdded)
```

Comment:
Do we still need this `else if (!mIsServiceAdded)` block?

The only change necessary might be:
```c++
#if OPENTHREAD_CONFIG_REFERENCE_DEVICE_ENABLE
        mSequenceNumber      = aConfig.mSequenceNumber;
#else
        mSequenceNumber      = aConfig.mSequenceNumber + 1; 
#endif
```

Thoughts? @librasungirl @edmont

Reply:
Agree, this also makes the trick.